### PR TITLE
remove consensus, extraneous parameter for update_view, repoint deps

### LINKED ISSUE #3640: [CX_HARDENING] - Delete Legacy Consensus
### What is this task and why do we need to work on it?

This is straightforward. Go in and delete the `consensus` module and clean up the compiler errors. This should include a check of the duplicated functions, however, before removal. In particular

- `update_view`
- `handle_quorum_proposal_validated`
- `create_and_send_proposal`

The purpose of these checks is to make sure that things like upgrade logic etc is included in the new code paths. We also need to rename `consensus2` to `consensus`.

### What work will need to be done to complete this task?

_No response_

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

- HotShot only has one `consensus` module.
- The dependency tasks are stable and deployed in all relevant environments.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by pls148**
Closes #3640

### This PR: 

This PR removes the consensus module (deprecated in favor of consensus2) and removes an extra parameter for update_view. Also removes references to the "dependency-tasks" feature. 

### This PR does not: 

This does NOT rename the consensus2 module to consensus, nor change any other types. That will come in a later PR.

### CODE REVIEW

**Comment by bfish713** [comment]
This looks good.  However I think now the CI is redundant.  The ci jobs for test and test-dependency-tasks are running the same tests now I think.  In a follow up can you confirm that the sets of tests are the same and remove one of them?

Future follow up is to pick out the now unused or misused types/functions and types in task-impl helpers (and other places)

**Review by bfish713 — APPROVED** [review]

**Review by ss-es — COMMENTED** [review]
I think the feature also appears in a few `Cargo.toml` files, and should be removed from there

For CI, you can also just delete the block called `test-dependency-tasks` in `.github/workflows/build-and-test.yml` (keeping the regular tests, which now always run the dependency tasks)

File: crates/testing/tests/tests_1/upgrade_task_with_proposal.rs [hunk 1]
```diff
@@ -4,7 +4,6 @@
 // You should have received a copy of the MIT License
 // along with the HotShot repository. If not, see <https://mit-license.org/>.
 
-#![cfg(feature = "dependency-tasks")]
 // TODO: Remove after integration of dependency-tasks
```
Comment (ss-es) [inline crates/testing/tests/tests_1/upgrade_task_with_proposal.rs:7]: I think you can remove this TODO

File: crates/testing/tests/tests_1/upgrade_task_with_vote.rs [hunk 2]
```diff
@@ -4,7 +4,6 @@
 // You should have received a copy of the MIT License
 // along with the HotShot repository. If not, see <https://mit-license.org/>.
 
-#![cfg(feature = "dependency-tasks")]
 // TODO: Remove after integration of dependency-tasks
```
Comment (ss-es) [inline crates/testing/tests/tests_1/upgrade_task_with_vote.rs:7]: this one too

**Review by ss-es — APPROVED** [review]
looks good to me!

I didn't look at the original PR in depth, but I think Brendon did so should be good to merge

**Pull request merged**